service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationRoutes.java

```diff
@@ -178,52 +180,72 @@ private void configureMasProcessing() {
         .to(END_POINT_RFD)
         .otherwise()
         // Off ramp if the Sufficient For Fast Tracking is null
-        .to(ENDPOINT_ACCESS_ERR)
+        .setProperty("offRampError", constant("Sufficiency cannot be determined."))
+        .setProperty("sourceRoute", constant("assessorError"))
+        .log("Assessor Error. Off-ramping claim")
+        .process(masAccessErrProcessor)
+        .to(ENDPOINT_OFFRAMP_ERROR)
         .end();
 
     String rfdRouteId = "mas-rfd";
+    String pdfFailError = "docUploadFailed";
     from(END_POINT_RFD)
         // input: MasAutomatedClaimPayload
         .routeId(rfdRouteId)
         .wireTap(ENDPOINT_AUDIT_WIRETAP)
         .onPrepare(auditProcessor(rfdRouteId, "Sufficient evidence for ready for decision."))
         // Upload PDF
+        .doTry()
         .to(ENDPOINT_UPLOAD_PDF)
-        // Check and update claim
+        .doCatch(BipException.class)
+        // Completion code needs the MasProcessingObject as the body.
+        .setBody(simple("${exchangeProperty.payload}"))
+        .setProperty("offRampError", constant(pdfFailError))
+        .setProperty("sourceRoute", constant(rfdRouteId))
+        .to(ENDPOINT_OFFRAMP_ERROR)
+        .stop()
+        .end() // End try
         .to(ENDPOINT_MAS_COMPLETE);
 
     // Call "Order Exam" in the absence of evidence .i.e Sufficient For Fast Tracking is "false"
     var orderExamRouteId = "mas-order-exam";
+    final String orderFailMessage = "examOrderFailed";
+    final String pdfFailMessage = "PDF upload failed after exam order requested.";
+
     from(ENDPOINT_ORDER_EXAM)
         // input: MasAutomatedClaimPayload
         .routeId(orderExamRouteId)
         .wireTap(ENDPOINT_AUDIT_WIRETAP)
         .onPrepare(
             auditProcessor(orderExamRouteId, "There is insufficient evidence. Ordering an exam"))
+        .doTry()
         .process(masOrderExamProcessor)
         .log("MAS Order Exam response: ${body}")
         // Upload PDF but catch errors since exam was ordered and continue
-        .doTry()
         .to(ENDPOINT_UPLOAD_PDF)
         .to(ENDPOINT_MAS_COMPLETE)
+        .doCatch(MasException.class)
+        // Body is still the Mas Processing object.
+        .setProperty("sourceRoute", constant(orderExamRouteId))
+        .setProperty("offRampError", constant(orderFailMessage))
+        .to(ENDPOINT_OFFRAMP_ERROR)
+        .stop() // Offramp and don't continue processing
         .doCatch(BipException.class)
         // Mas Complete Processing code expects this to be the body of the message
         .setBody(simple("${exchangeProperty.payload}"))
+        // Wiretap will cause no code to execute after the end of the try. Intentional here.
```

Comment:
I wanted to not rely on the wiretap side-effect behavior here which stops processing, but i'd have to make an entirely new route just to contain the wiretap in a *non* offramp scenario and somehow pass the onprepare the right variables differently. 
Due to time I commented that I am relying on the strange behavior here. (This case execution did not change in this ticket, i just made endDoCatch() end which is more correct in this case even with wiretap involved).